plat/rockchip/rk3399/drivers/dp/cdn_dp.c

```diff
@@ -41,3 +44,46 @@ __asm__(
 	".popsection\n"
 );
 
+static uint64_t *hdcp_key_pdata;
+static struct cdn_dp_hdcp_key_1x key;
+
+int hdcp_handler(struct cdn_dp_hdcp_key_1x *key);
+
+uint64_t dp_hdcp_ctrl(uint64_t type)
+{
+	switch (type) {
+	case HDCP_KEY_DATA_START_TRANSFER:
+		memset(&key, 0x00, sizeof(key));
+		hdcp_key_pdata = (uint64_t *)&key;
+		return 0;
+	case HDCP_KEY_DATA_START_DECRYPT:
+		if (hdcp_key_pdata == (uint64_t *)(&key + 1))
```

Comment:
@Caesar-github 
Apologies for the slow response.
I don't think that the __align(8) will work - perhaps a CASSERT on the size of cdn_dp_hdcp_key_1x to be a multiple of 6x64 bits; this would warn you if that struct changed in size to not be aligned with 6x64 bits.